beetsplug/vocadb/__init__.py

```diff
@@ -0,0 +1,950 @@
+                album_id,
+            )
+            return None
+        self._log.debug("Searching for album {0}", album_id)
+        album: Album | None = self.client._get(
+            relative_path=f"albums/{album_id}",
+            params={
+                "lang": self.instance_config.language,
+                "fields": "Artists,Discs,Tags,Tracks,WebLinks",
+                "songFields": SONG_FIELDS,
+            },
+            type=Album,
+        )
+        return (
+            self.album_info(album, search_lang=self.instance_config.language)
+            if album
+            else None
+        )
+
+    @override
+    def track_for_id(self, track_id: str) -> TrackInfo | None:
+        if not track_id.isnumeric():
+            self._log.debug(
+                "Skipping non-{0} singleton: {1}",
+                self.data_source,
+                track_id,
+            )
+            return None
+        self._log.debug("Searching for track {0}", track_id)
+        track: Song | None = self.client._get(
+            relative_path=f"songs/{track_id}",
+            params={
+                "lang": self.instance_config.language,
+                "fields": SONG_FIELDS,
+            },
+            type=Song,
+        )
+        return (
+            self.track_info(track, search_lang=self.instance_config.language)
+            if track
+            else None
+        )
+
+    def album_info(
+        self, release: Album, search_lang: str | None = None
+    ) -> AlbumInfo:
+        if not release.discs:
+            release.discs = [
+                Disc(
+                    disc_number=i + 1, name="CD", media_type=DiscMediaType.AUDIO
+                )
+                for i in range(
+                    max(track.disc_number for track in release.tracks)
+                )
+            ]
+        ignored_discs: set[int] = set()
+        disc: Disc
+        for disc in release.discs:
+            disc_number: int = disc.disc_number
+            if (
+                disc.media_type == DiscMediaType.VIDEO
+                and config["match"]["ignore_video_tracks"].get(bool)
+                or not release.tracks
+            ):
+                ignored_discs.add(disc_number)
+                continue
+            disc.total = max(
+                {
+                    track.track_number
+                    for track in release.tracks
+                    if track.disc_number == disc_number
+                }
+            )
+
+        va: bool = release.disc_type == DiscTypes.COMPILATION
+        album: str | None = release.name
+        album_id: str | None = str(release.id)
+        artist_categories: ArtistsByCategories
+        artist: str
+        artist_categories, artist = self.get_artists(
+            release.artists,
+            include_featured_artists=self.instance_config.include_featured_album_artists,
+            comp=va,
+        )
+        if artist == VA_NAME:
+            va = True
```

Comment:
(doesn't really concern the changes in this pr but...)
We could remove setting the `va` variable above and instead use

```suggestion
        album: str | None = release.name
        album_id: str | None = str(release.id)
        artist_categories: ArtistsByCategories
        artist: str
        artist_categories, artist = self.get_artists(
            release.artists,
            include_featured_artists=self.instance_config.include_featured_album_artists,
            comp=va,
        )
        va: bool = release.disc_type == DiscTypes.COMPILATION or artist == VA_NAME
```

Reply:
but the get_artists need the comp parameter. How about this?
```Python
artist_categories, artist = self.get_artists(
    release.artists,
    include_featured_artists=self.instance_config.include_featured_album_artists,
    comp=release.disc_type == DiscTypes.COMPILATION,
)
va: bool = artist == VA_NAME
```

Reply:
Oh yeah, I forgot `get_artists` uses it... I don't think that suggestion works either, since the `artist` variable may include featured artists.

I guess one way would be to replace it with
`va = va or artist == VA_NAME`

or just leave it as is, but I'll leave the decision up to you.